source/parse.h

```diff
@@ -4259,6 +4284,11 @@ class parser
                 n->for_with_in = true;
             }
 
+            if (!done() && curr().type() == lexeme::Semicolon) {
+                error("for..do loop body shall not end with a semicolon");
+                return {};
```

Comment:
"a loop body may not be followed by a semicolon (empty statements are not allowed)"